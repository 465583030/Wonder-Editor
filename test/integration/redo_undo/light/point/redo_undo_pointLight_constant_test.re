open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: pointLight constant", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => sandbox := createSandbox());
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    let _changeConstant = value => {
      let currentGameObjectPointLightComponent =
        GameObjectTool.getCurrentGameObjectPointLightComponent();
      let component =
        BuildComponentTool.buildPointLight(
          currentGameObjectPointLightComponent,
        );

      let constantDomIndex = MainEditorLightTool.getConstantDomIndex();

      BaseEventTool.triggerComponentEvent(
        component,
        MainEditorLightTool.triggerLightComponentChangeEvent(
          constantDomIndex,
          value,
        ),
      );
      BaseEventTool.triggerComponentEvent(
        component,
        MainEditorLightTool.triggerLightComponentBlurEvent(
          constantDomIndex,
          value,
        ),
      );
    };

    let _simulateTwiceChangeConstant = () => {
      let value1 = 10.1;
      let value2 = 12.12;

      MainEditorLightTool.setLightTypeToBePointLight();

      _changeConstant(value1);
      _changeConstant(value2);
    };

    let _beforeEach = () => {
      MainEditorSceneTool.initStateWithJob(
        ~sandbox,
        ~noWorkerJobRecord=
          NoWorkerJobConfigToolEngine.buildNoWorkerJobConfig(),
        (),
      );

      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;

      MainEditorSceneTool.createDefaultScene(sandbox, () =>
        MainEditorSceneTool.getDirectionLightInDefaultScene
        |> StateLogicService.getEngineStateToGetData
        |> GameObjectTool.setCurrentSceneTreeNode
      );
      DirectorToolEngine.prepareAndInitAllEnginState();
    };

    RedoUndoTool.testRedoUndoTwoStep(
      sandbox,
      "prepare first step: set currentSceneTreeNode",
      (_simulateTwiceChangeConstant, _beforeEach, () => ()),
      BuildComponentForRedoUndoTool.buildPointLight,
    );
  });