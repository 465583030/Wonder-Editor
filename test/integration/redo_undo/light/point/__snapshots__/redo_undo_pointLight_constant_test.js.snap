// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: pointLight constant prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from second to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-point-light"
>
  <article
    className="wonder-material-color"
  >
    <div
      className=""
    >
      <span
        className=""
      >
        color : 
      </span>
      <span
        className=""
      >
        #ffffff
      </span>
      <button
        className=""
        onClick={[Function]}
      >
        pick color
      </button>
    </div>
  </article>
  <div
    className="light-intensity"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Intensity : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </article>
    </article>
  </div>
  <div
    className="light-constant"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Constant : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="10.1"
        />
      </article>
    </article>
  </div>
  <div
    className="light-linear"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Linear : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.07"
        />
      </article>
    </article>
  </div>
  <div
    className="light-quadratic"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Quadratic : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.017"
        />
      </article>
    </article>
  </div>
  <div
    className="light-range"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Range : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="65"
        />
      </article>
    </article>
  </div>
</article>
`;

exports[`redo_undo: pointLight constant prepare first step: set currentSceneTreeNode test redo operate test redo two step undo step which from second to zero,redo step which from zero to second 1`] = `
<article
  className="wonder-point-light"
>
  <article
    className="wonder-material-color"
  >
    <div
      className=""
    >
      <span
        className=""
      >
        color : 
      </span>
      <span
        className=""
      >
        #ffffff
      </span>
      <button
        className=""
        onClick={[Function]}
      >
        pick color
      </button>
    </div>
  </article>
  <div
    className="light-intensity"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Intensity : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </article>
    </article>
  </div>
  <div
    className="light-constant"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Constant : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="12.12"
        />
      </article>
    </article>
  </div>
  <div
    className="light-linear"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Linear : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.07"
        />
      </article>
    </article>
  </div>
  <div
    className="light-quadratic"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Quadratic : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.017"
        />
      </article>
    </article>
  </div>
  <div
    className="light-range"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Range : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="65"
        />
      </article>
    </article>
  </div>
</article>
`;

exports[`redo_undo: pointLight constant prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-point-light"
>
  <article
    className="wonder-material-color"
  >
    <div
      className=""
    >
      <span
        className=""
      >
        color : 
      </span>
      <span
        className=""
      >
        #ffffff
      </span>
      <button
        className=""
        onClick={[Function]}
      >
        pick color
      </button>
    </div>
  </article>
  <div
    className="light-intensity"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Intensity : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </article>
    </article>
  </div>
  <div
    className="light-constant"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Constant : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="12.12"
        />
      </article>
    </article>
  </div>
  <div
    className="light-linear"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Linear : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.07"
        />
      </article>
    </article>
  </div>
  <div
    className="light-quadratic"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Quadratic : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.017"
        />
      </article>
    </article>
  </div>
  <div
    className="light-range"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Range : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="65"
        />
      </article>
    </article>
  </div>
</article>
`;

exports[`redo_undo: pointLight constant prepare first step: set currentSceneTreeNode test undo operate test undo one step step which from second to first 1`] = `
<article
  className="wonder-point-light"
>
  <article
    className="wonder-material-color"
  >
    <div
      className=""
    >
      <span
        className=""
      >
        color : 
      </span>
      <span
        className=""
      >
        #ffffff
      </span>
      <button
        className=""
        onClick={[Function]}
      >
        pick color
      </button>
    </div>
  </article>
  <div
    className="light-intensity"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Intensity : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </article>
    </article>
  </div>
  <div
    className="light-constant"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Constant : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="10.1"
        />
      </article>
    </article>
  </div>
  <div
    className="light-linear"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Linear : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.07"
        />
      </article>
    </article>
  </div>
  <div
    className="light-quadratic"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Quadratic : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.017"
        />
      </article>
    </article>
  </div>
  <div
    className="light-range"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Range : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="65"
        />
      </article>
    </article>
  </div>
</article>
`;

exports[`redo_undo: pointLight constant prepare first step: set currentSceneTreeNode test undo operate test undo two step step which from second to zero 1`] = `
<article
  className="wonder-point-light"
>
  <article
    className="wonder-material-color"
  >
    <div
      className=""
    >
      <span
        className=""
      >
        color : 
      </span>
      <span
        className=""
      >
        #ffffff
      </span>
      <button
        className=""
        onClick={[Function]}
      >
        pick color
      </button>
    </div>
  </article>
  <div
    className="light-intensity"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Intensity : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </article>
    </article>
  </div>
  <div
    className="light-constant"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Constant : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </article>
    </article>
  </div>
  <div
    className="light-linear"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Linear : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.07"
        />
      </article>
    </article>
  </div>
  <div
    className="light-quadratic"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Quadratic : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.017"
        />
      </article>
    </article>
  </div>
  <div
    className="light-range"
  >
    <article
      className="wonder-light-base"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Range : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="65"
        />
      </article>
    </article>
  </div>
</article>
`;
